### PR TITLE
Cw 78 ethereum

### PR DESCRIPTION
**Pull request by OmarHatem28**
Issue Number (if Applicable): Fixes #

# Description

Adding Ethereum Wallet.

- [x] Create Ethereum Wallet
- [x] Restore wallet
- [x] Connect to Eth node
- [x] Receive Eth
- [x] Send Eth
- [x] Show Transactions

# Pull Request - Checklist  

- [x] Initial Manual Tests Passed
- [ ] Double check modified code and verify it with the feature/task requirements
- [x] Formate code
- [ ] Look for code duplication
- [x] Clear naming for variables and methods

### CODE REVIEW

**Review by fossephate — APPROVED** [review]
Some minor nit-picks and some notes in the comments but I didn't spot anything major.

File: cw_ethereum/lib/ethereum_client.dart [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+      _client = Web3Client(node.uri.toString(), _httpClient);
+
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  void setListeners(EthereumAddress userAddress, Function() onNewTransaction) async {
+    // _client?.pendingTransactions().listen((transactionHash) async {
```
[the remark this answers is not in the document]
Comment (OmarHatem28) [reply cw_ethereum/lib/ethereum_client.dart:32]: yeah we are delaying that until we get our own node to listen to transactions

File: .gitignore
[the remark this answers is not in the document]
Comment (OmarHatem28) [reply .gitignore:1]: `.secrets.g.dart` in the ethereum package is indeed ignore, and is generated via the generate secrets scripts

[the remark this answers is not in the document]
Comment (fossephate) [reply .gitignore:1]: ah, my local `secrets.dart` file is probably erroneous then

**Pull request merged**